databasetest/databasetest.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package databasetest provides an ephemeral database server for integration testing.
+// It's designed to be used in code that needs to work inside and outside google.
+package databasetest
```

Comment:
(deep thoughts warning) like, dude, is there anything about this that's actually specific to databases? Or is it just a higher level interface to docker test containers?

Reply:
Its definitely the latter, wasn't sure if we wanted to push that usecase or not

Reply:
Seems like just a matter of time until this use case (non-databases) arises. If we can address it now just by using a different name, maybe it's worth it.

Reply:
I think the "run a container" bit is generalizable, but the port mapping and "wait for ready" seems pretty specialized to databases.

Reply:
If we want to change, what would be a good name? Its a wrapper for `dockertest`, so I wouldn't want to use that name. I was thinking maybe `testcontainer` but its a little too close to https://golang.testcontainers.org/

`testdockerdeps`? Not a fan of that either

Reply:
I think I would rename `Driver` to `Service`

Reply:
Changes made, let me know what you think

Reply:
> Wouldn't port mapping and wait for ready also apply to pretty much any IP-based dependency? (Say a message queue simulator, some 3p app, shim for some GCP dependency for local testing)

+1, any server with non-negligible startup time would need this, I think.